frontend/pkg/frontend/frontend.go

```diff
@@ -287,7 +361,7 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 
 	var hcpCluster *api.HCPOpenShiftCluster
 	var csResp *cmv1.ClusterGetResponse
-	if doc.ClusterID != "" {
+	if doc.ClusterID != "" { // redundant if check ? at this point doc.ClusterID can't be "", otherwise error raised.
```

Comment:
This is likely old code before a lot of the error handling above it was defined. +1 that it can go away

Reply:
Actually i take that back -- the reason is there is if we dont find a cluter document, it means we don't have a cluster and clusterID would be empty. So we dont want to call CS to get the cluster if it doesnt exist yet. Its logic to skip execution which would fail

Reply:
True. Thanks! IMHO The fact that we use the same functions for both create and update is making the code a bit more complex than it could be as we need to think about both cases at the same time.